module/polls/helper.js

```diff
@@ -0,0 +1,631 @@
+                                ]
+                            }
+                    },
+                    [
+                      "value"
+                    ]
+                )
+
+                pollCreationForm = [...pollCreationForm[0].value,
+                                    ...pollCreationForm[1].value]
+                
+                let unicodes = await mediaFilesHelper.mediaFileDocuments
+                ( 
+                    {
+                        status: "active"
+                    }
+                )
+
+                let emoticons = [];
+                let gestures = [];
+
+                if (unicodes.length > 0) {
+                    unicodes.forEach( unicode => {
+                         if (unicode.type == messageConstants.common.EMOJI) {
+                            emoticons.push(unicode);
+                         }
+                         else if (unicode.type == messageConstants.common.GESTURE){
+                             gestures.push(unicode);
+                         }
+                    });
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLL_CREATION_FORM_FETCHED,
+                    data: {
+                          form: pollCreationForm,
+                          emoticons: emoticons,
+                          gestures: gestures
+                    }
+                });
+
+            } catch (error) {
+                 return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+    /**
+     * Create poll
+     * @method
+     * @name create
+     * @param {Object} pollData - poll creation data
+     * @param {String} userId - userId
+     * @param {String} appName - Name of the app
+     * @returns {String} - Sharable link .
+     */
+
+    static create(pollData= {}, userId= "", appName= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (Object.keys(pollData).length == 0) {
+                    throw new Error (messageConstants.apiResponses.POLL_DATA_REQUIRED);
+                }
+
+                if (userId == "") {
+                    throw new Error (messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                if (appName == "") {
+                    throw new Error (messageConstants.apiResponses.APP_NAME_FIELD_REQUIRED);
+                }
+
+                let pollDocument = {
+                    name: pollData.name,
+                    creator: pollData.creator,
+                    createdBy: userId,
+                    startDate: new Date(),
+                    endDate: new Date(new Date().setDate(new Date().getDate() + pollData.endDate)),
+                    metaInformation: pollData.metaInformation,
```

Comment:
@deepakr123 - How is it determined what goes into metaInformation ?

Reply:
resolved